### PR TITLE
[ML] Add deployment threading details and memory usage to telemetry

### PR DESCRIPTION
**Pull request by davidkyle**
Creates a new `memory` section of the usage reporting and adds the following fields. The values come from the [ML Memory API](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-ml-memory.html)

- memory.anomaly_detectors_memory_bytes          -> Total memory used by AD jobs
- memory.data_frame_analytics_memory_bytes      -> Total memory used by DFA jobs
- memory.pytorch_inference_memory_bytes            -> Total memory used by deployed PyTorch models
- memory.total_used_memory_bytes                         -> Total memory used by ML. Is the sum of the above 3 fields plus the native process overheads

The deployment threads count and allocation count is now added the `stats_by_model` section  

- inference.deployments.stats_by_model.$LIST_INDEX.num_allocations
- inference.deployments.stats_by_model.$LIST_INDEX.num_threads

### CODE REVIEW

**Review by prwhelan — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningFeatureSetUsage.java [hunk 1]
```diff
@@ -31,11 +31,13 @@ public class MachineLearningFeatureSetUsage extends XPackFeatureSet.Usage {
...
+    public static final String MEMORY_FIELD = "memory";
 
     private final Map<String, Object> jobsUsage;
     private final Map<String, Object> datafeedsUsage;
     private final Map<String, Object> analyticsUsage;
     private final Map<String, Object> inferenceUsage;
+    private final Map<String, Object> memoryUsage;
     private final int nodeCount;
 
     public MachineLearningFeatureSetUsage(
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningFeatureSetUsage.java:43]: Yes we do. Thanks

**Comment by davidkyle** [comment]
@elasticmachine update branch

**Comment by arisonl** [comment]
@davidkyle can we please add the RAM and CPU utilization as described here? https://github.com/elastic/ml-team/issues/1343

**Comment by thecoop** [comment]
@davidkyle this change must be backported to 8.x, there is currently inconsistency between 8.x and main

**Pull request merged**